packages/sidebar/src/Sidebar.tsx

```diff
@@ -36,67 +36,85 @@ export const PureSidebar: React.FC<ISidebarProps> = ({
   isRTL = false,
   minWidth = 30,
   maxWidth = 400,
+  rollWidth = 10,
+  params: {
+    collapsed = false,
+    width: propsWidth = maxWidth,
+  } = {},
   navWidth = 90,
   separatorWidth = 10,
+  onChangeWidth = () => {},
 }) => {
   const themeContext = useContext(ThemeContext);
 
-  const [width, setWidth] = useState(maxWidth);
+  const [width, setWidth] = useState(collapsed ? rollWidth : propsWidth);
   const [animate, setAnimate] = useState(false);
-  const [arrowToRight, setArrowToRight] = useState(false);
+  const [arrowToRight, setArrowToRight] = useState(collapsed);
   const [antiSelectLayer, setAntiSelectLayer] = useState(false);
   const [offsetLeft, changeOffsetLeft] = useState(0);
 
   const sidebarRef = useRef<HTMLDivElement>(null);
   const observerRef: React.MutableRefObject<ResizeObserver | undefined> = useRef();
   const clickXRef = useRef(0);
-  const widthRef = useRef(maxWidth);
+  const widthRef = useRef(propsWidth);
+
+  useEffect(() => {
+    setArrowToRight(width < minWidth);
+  }, [minWidth, propsWidth, width]);
+
+  useEffect(() => {
+    setWidth(collapsed ? rollWidth : propsWidth);
+    setArrowToRight(collapsed);
+  }, [collapsed, propsWidth]);
 
   const handleMouseMove = useCallback(
     (e) => {
       const { clientX: currentX } = e;
 
       const newWidth = isRTL
-        ? width - (currentX - clickXRef.current)
-        : width + (currentX - clickXRef.current);
+        ? Number(width) - Number(currentX - clickXRef.current)
```

Comment:
Why you add Number? This values already numbers.

Reply:
yep. just add type **MouseEvent**  for **e** variable